doc/developers/contributing.rst

```diff
@@ -268,100 +249,115 @@ If any of the above seems like magic to you, then look up the `Git documentation
-  one is preceded by a keyword). Upon merging, those issues/PRs will
-  automatically be closed by GitHub. If your pull request is simply
-  related to some other issues/PRs, create a link to them without using
-  the keywords (e.g., ``See also #1234``).
-
-* All public methods should have informative docstrings with sample
-  usage presented as doctests when appropriate.
-
-* Please prefix the title of your pull request with ``[MRG]`` if the
-  contribution is complete and should be subjected to a detailed review.
-  Two core developers will review your code and change the prefix of the pull
-  request to ``[MRG + 1]`` and ``[MRG + 2]`` on approval, making it eligible
-  for merging. An incomplete contribution -- where you expect to do more
-  work before receiving a full review -- should be prefixed ``[WIP]`` (to
-  indicate a work in progress) and changed to ``[MRG]`` when it matures.
-  WIPs may be useful to: indicate you are working on something to avoid
-  duplicated work, request broad review of functionality or API, or seek
-  collaborators. WIPs often benefit from the inclusion of a
-  `task list
-  <https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments>`_
-  in the PR description.
-
-* All other tests pass when everything is rebuilt from scratch. On
-  Unix-like systems, check with (from the toplevel source folder)::
-
-    $ make
-
-* When adding additional functionality, provide at least one example script
-  in the ``examples/`` folder. Have a look at other examples for reference.
-  Examples should demonstrate why the new functionality is useful in
-  practice and, if possible, compare it to other methods available in
-  scikit-learn.
-
-* Documentation and high-coverage tests are necessary for enhancements to be
-  accepted. Bug-fixes or new features should be provided with
-  `non-regression tests
-  <https://en.wikipedia.org/wiki/Non-regression_testing>`_. These tests
-  verify the correct behavior of the fix or feature. In this manner, further
-  modifications on the code base are granted to be consistent with the
-  desired behavior. For the case of bug fixes, at the time of the PR, the
-  non-regression tests should fail for the code base in the master branch
-  and pass for the PR code.
-
-* At least one paragraph of narrative documentation with links to
-  references in the literature (with PDF links when possible) and
-  the example. For more details on writing and building the
-  documentation, see the :ref:`contribute_documentation` section.
-
-* The documentation should also include expected time and space complexity
-  of the algorithm and scalability, e.g. "this algorithm can scale to a
-  large number of samples > 100000, but does not scale in dimensionality:
-  n_features is expected to be lower than 100".
+In order to ease the reviewing process, we recommend that your contribution
+complies with the following rules before submitting a pull request. The
```

Comment:
I don't agree that this should qualify the submission of a pull request. But it should be considered a draft until it complies

Reply:
I agree, these requirements are more for a PR to be accepted. For instance you might not understand why some test is failing and submitting your PR allows other devs to take a look and give some help.

You could advise to tag the PR with [WIP] until it does not fulfill all the items below.

Reply:
Good point